dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -947,50 +947,26 @@ public List<BlockWorkerInfo> getLostWorkers() throws IOException {
    *
    * @return the info of all workers
    */
-  public List<BlockWorkerInfo> getAllWorkers() throws IOException {
+  public WorkerClusterView getAllWorkers() throws IOException {
     // TODO(lucy) once ConfigHashSync reinit is gotten rid of, will remove the blockReinit
     // guard altogether
     try (ReinitBlockerResource r = blockReinit()) {
       // Use membership mgr
       if (mMembershipManager != null && !(mMembershipManager instanceof MasterMembershipManager)) {
-        List<BlockWorkerInfo> liveWorkers = mMembershipManager.getLiveMembers().stream()
-            .map(w -> new BlockWorkerInfo(w.getIdentity(), w.getAddress(), w.getCapacityBytes(),
-                w.getUsedBytes(), true)).collect(toList());
-        List<BlockWorkerInfo> lostWorkers = mMembershipManager.getFailedMembers().stream()
-            .map(w -> new BlockWorkerInfo(
-                w.getIdentity(), w.getAddress(), w.getCapacityBytes(), w.getUsedBytes(),
-                false)).collect(toList());
-        // avoid duplicate elements in list
-        return combineAllWorkers(liveWorkers, lostWorkers);
+        return mMembershipManager.getAllMembers();
```

Comment:
I was confused why we are taking trouble to combine a list of live workers with a list of lost workers to get the list of all workers, while all this time there is a `getAllMembers` method to do exactly this.

I think this change has a subtle difference, as the `combineAllWorkers` method used to de-duplicate workers by their net address. Now `mMembershipManager.getAllMembers` does so by checking the worker IDs. But workers shouldn't have conflicting net addresses either so it's no big problem IMO.

Reply:
Agreed we can use `mMembershipManager.getAllMembers` given the subtle difference. But the property that `workers shouldn't have conflicting net addresses` is not guaranteed IMO? I suggest we double check that in the worker register code (in a separate PR if necessary). At least if some workers are using dup IDs or addresses we should check somewhere and log some warnings, so that people can realize at all.

Reply:
> workers shouldn't have conflicting net addresses is not guaranteed

Right. So there used to be some potential bugs due to conflicting worker addresses. But this is not relevant any more. Depending on net addresses to differentiate workers is not reliable, hence the introduction of worker IDs.

> we double check that in the worker register code  

I think Lucy did exactly that in https://github.com/Alluxio/alluxio/pull/18454

Reply:
@jiacheliu3   @dbw9580  actually if one worker register with workerid1 and then remove its workeridentity file and restart, it will register as workerid2,  if no one removes workerid1 key from etcd, getAllMembers will have both worker entity bearing same addresses, but that's not sth code base should be guarding against, its a deployment issue. getAllMembers could be thought of as all the distinct members forming the ring.